### PR TITLE
Add several PR jobs to CI on GKE.

### PR DESCRIPTION
**Pull request by spxtr**
I split the PR jobs into their own files so it's less confusing.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/test-infra/494)

### CODE REVIEW

File: ciongke/cmd/hook/main.go [hunk 1]
```diff
@@ -45,6 +45,38 @@ var (
 )
 
 var defaultJenkinsJobs = map[string][]JenkinsJob{
+	"kubernetes/kubernetes": {
```
Comment (ixdy) [inline ciongke/cmd/hook/main.go]: should this be in a config file instead of in code?

Comment (spxtr) [inline ciongke/cmd/hook/main.go]: Yes, eventually it'll either be in a ConfigMap or in a file in the repo.

**Comment by ixdy** [comment]
How will we avoid triggering all open PRs when this is enabled?

**Comment by ixdy** [comment]
oh, probably cause this is a webhook, not polling. right.

**Comment by spxtr** [comment]
Yep, it will start responding to new events and won't try to rebuild the world.

**Comment by rmmh** [comment]
I really dislike the duplication in the YAML.

I know the combined YAML can be hard to follow, but it still seems better than repeating lots of magic sections in *different* files.

**Comment by ixdy** [comment]
yeah... also you didn't split out everything, so now it's like the worst of both worlds.

**Comment by spxtr** [comment]
OK, I kept the templating. PTAL

**Comment by ixdy** [comment]
This appears to have caused a bunch of unrelated changes, e.g. on `kubernetes-pull-build-test-e2e-gke`, maxConcurrentPerNode dropped to 1, the overall timeout changed from 90m to 600m, etc.

**Comment by spxtr** [comment]
Our nodes only have one executor so having maxConcurrentPerNode at 2 doesn't make sense. I guess I'll set the timeout back, I don't remember why I changed that.

**Comment by ixdy** [comment]
This still changes the timeout (from 90m to 80m), but I'm fine with that.

LGTM

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml [hunk 2]
```diff
@@ -478,143 +401,6 @@
               echo "Exiting with code: ${{rc}}"
               exit ${{rc}}
             fi
-
-      - build-test-federation-e2e-gce: # kubernetes-pull-build-test-federation-e2e-gce
-          status-context: Federation GCE e2e
-          max-total: 12
-          only-trigger-phrase: true  # This test should run only if explicitly triggered.
```
Comment (nikhiljindal) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: I see that we are not setting this in the new file.
Based on the PR description, it seems like the intention was just moving the config, not changing it.
@spxtr  Is the change intentional?

**Comment by spxtr** [comment]
Yes, that logic is in `ciongke` now.

File: jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml (hunk 2 above)
Comment (spxtr) [inline jenkins/job-configs/kubernetes-jenkins-pull/kubernetes-pull.yaml]: Yes, that logic is in ciongke now.
